OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -133,13 +135,13 @@ public DecisionService(Bucketer bucketer, IErrorHandler errorHandler, UserProfil
             {
                 try
                 {
-                    Dictionary<string, object> userProfileMap = UserProfileService.Lookup(userId);
+                    Dictionary<string, object> userProfileMap = UserProfileService.Lookup(user.GetUserId());
                     if (userProfileMap != null && UserProfileUtil.IsValidUserProfileMap(userProfileMap))
                     {
                         userProfile = UserProfileUtil.ConvertMapToUserProfile(userProfileMap);
-                        variationResult = GetStoredVariation(experiment, userProfile, config);
-                        reasons += variationResult.DecisionReasons;
-                        if (variationResult.ResultObject != null) return variationResult.SetReasons(reasons);
+                        decisionVariation = GetStoredVariation(experiment, userProfile, config);
+                        reasons += decisionVariation?.DecisionReasons;
+                        if (decisionVariation?.ResultObject != null) return decisionVariation?.SetReasons(reasons);
```

Comment:
Do we need all these null checking?  If it's null, we can lose reasons in line 144.

Reply:
I have removed all null checking for it. Result `class` will always be non-nullable when returned `NullResult` or `NewResult`.